lib/declare_schema/model/habtm_model_shim.rb

```diff
@@ -4,22 +4,25 @@ module DeclareSchema
   module Model
     class HabtmModelShim
       class << self
-        def from_reflection(refl)
-          new(refl.join_table, [refl.foreign_key, refl.association_foreign_key],
-                               [refl.active_record.table_name, refl.class_name.constantize.table_name])
+        def from_reflection(reflection, connection:)
+          new(reflection.join_table,
+              [reflection.foreign_key, reflection.association_foreign_key],
+              [reflection.active_record.table_name, reflection.class_name.constantize.table_name],
+              connection: connection)
         end
       end
 
-      attr_reader :join_table, :foreign_keys, :parent_table_names
+      attr_reader :join_table, :foreign_keys, :parent_table_names, :connection
 
-      def initialize(join_table, foreign_keys, parent_table_names)
+      def initialize(join_table, foreign_keys, parent_table_names, connection:)
         foreign_keys.is_a?(Array) && foreign_keys.size == 2 or
           raise ArgumentError, "foreign_keys must be <Array[2]>; got #{foreign_keys.inspect}"
         parent_table_names.is_a?(Array) && parent_table_names.size == 2 or
           raise ArgumentError, "parent_table_names must be <Array[2]>; got #{parent_table_names.inspect}"
         @join_table = join_table
         @foreign_keys = foreign_keys.sort # Rails requires these be in alphabetical order
         @parent_table_names = @foreign_keys == foreign_keys ? parent_table_names : parent_table_names.reverse # match the above sort
+        @connection = connection
```

Comment:
What is the point of passing the connection into this class?

I'm not really seeing where / how its getting used

Reply:
It was strictly for the `attr_reader :connection`. But good call to flag it because that was no longer used by the migrator.